src/communication/index.ts

```diff
@@ -0,0 +1,29 @@
+import { SDKMessageToPayload, SDKMessageIds, SentSDKMessage, RequestId } from '../types';
+
+const sendMessageToInterface = <T extends SDKMessageIds, D = SDKMessageToPayload[T]>(
+  messageId: T,
+  data: D,
+  requestId?: RequestId,
+): SentSDKMessage<T, D> => {
+  if (!requestId) {
+    if (typeof window !== 'undefined') {
+      requestId = Math.trunc(window?.performance.now());
+    } else {
+      requestId = Math.trunc(Date.now());
+    }
+  }
+  const message = {
+    messageId,
+    requestId,
+    data,
+  };
+
+  if (typeof window !== 'undefined') {
+    window.parent.postMessage(message, '*');
```

Comment:
shouldn't we throw here, as we currently provide no alternative for this ... in the future this might change to support other communication channels (e.g. for mobile apps)

Reply:
I think it makes sense. Maybe we should check that window exists at the initialization step